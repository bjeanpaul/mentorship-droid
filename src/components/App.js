import React from 'react-native';
import { Provider } from 'react-redux';

import configureStore from '../stores/configureStore';
const store = configureStore();


console.log('----------------------------')
import { login } from '../mentor/MentorActions';
store.dispatch(login('admin', '123'));
console.log('----------------------------')


import RootNavigator from './RootNavigator';



const App = function App() {
  return (
    <Provider store={store}>
        <RootNavigator />
    </Provider>
  );
};

export default App;
