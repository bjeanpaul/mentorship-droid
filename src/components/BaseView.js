import React from 'react';
import { View, StyleSheet, StatusBar } from 'react-native';
import colors from 'src/constants/colors';


const styles = StyleSheet.create({
  view: {
    backgroundColor: colors.DEFAULT_BG,
    flex: 1,
  },
});

const BaseView = ({
  children,
  style,
  statusBarBackgroundColor = colors.DEFAULT_STATUS_BG,
}) => (
  <View style={[styles.view, style]}>
    { statusBarBackgroundColor
      ? <StatusBar backgroundColor={statusBarBackgroundColor} />
      : null
    }
    {children}
  </View>
);
BaseView.propTypes = {
  statusBarBackgroundColor: React.PropTypes.string,
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};
export default BaseView;
