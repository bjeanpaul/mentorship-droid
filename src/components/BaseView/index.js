import React from 'react';
import { View } from 'react-native';

import styles from './styles';


const BaseView = ({
  children,
  style,
}) => (
  <View style={[styles.view, style]}>
    {children}
  </View>
);


BaseView.propTypes = {
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};


export default BaseView;
