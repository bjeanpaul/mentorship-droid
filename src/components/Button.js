import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
