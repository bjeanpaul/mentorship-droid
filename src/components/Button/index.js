import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
