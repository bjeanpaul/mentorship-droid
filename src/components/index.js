import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
