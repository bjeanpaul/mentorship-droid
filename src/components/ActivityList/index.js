import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
