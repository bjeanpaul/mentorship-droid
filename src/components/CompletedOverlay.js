import React, { PropTypes } from 'react';
import { View, StyleSheet, Image } from 'react-native';
import { BaseView, Text, Button } from 'src/components';

import COLORS from 'src/constants/colors';
import IMAGE_DISMISS_ICON from 'app/assets/close-light.png';
import IMAGE_DONE from 'app/assets/done-icon.png';


const styles = StyleSheet.create({
  baseView: {
    backgroundColor: COLORS.OVERLAY_BG,
  },
  content: {
    flex: 0.7,
    justifyContent: 'center',
  },
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    flex: 1,
    height: 80,
    width: 80,
  },
  footer: {
    flex: 0.3,
    justifyContent: 'flex-end',
  },
  dismissButton: {
    position: 'absolute',
    left: 16,
    top: 16,
  },
  title: {
    fontSize: 20,
    color: COLORS.OVERLAY_TEXT,
  },
  message: {
    fontSize: 24,
    color: COLORS.OVERLAY_MESSAGE,
  },
  footerText: {
    padding: 16,
    fontSize: 14,
  },
});

// TODO: Dismiss Action
const CompletedOverlay = ({
  title,
  message,
  buttonLabel,
  buttonHandlePress,
  bottomText,
  bottomTextHandlePress,
}) => (
  <BaseView
    style={styles.baseView}
    statusBarBackgroundColor={COLORS.OVERLAY_STATUS_BG}
  >
    <View style={styles.content}>
      <Image
        style={styles.dismissButton}
        source={IMAGE_DISMISS_ICON}
      />
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={IMAGE_DONE}
        />
      </View>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.message}>{message}</Text>
    </View>

    <View style={styles.footer}>
      <Button
        theme="white"
        handlePress={buttonHandlePress}
        label={buttonLabel}
      />
      {bottomText
        ? <Text style={styles.footerText} onPress={() => bottomTextHandlePress}>
          {bottomText}
        </Text>
        : null
      }
    </View>
  </BaseView>
);

CompletedOverlay.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  buttonLabel: PropTypes.string.isRequired,
  buttonHandlePress: PropTypes.func.isRequired,
  bottomText: PropTypes.string,
  bottomTextHandlePress: PropTypes.func,
};
export default CompletedOverlay;
