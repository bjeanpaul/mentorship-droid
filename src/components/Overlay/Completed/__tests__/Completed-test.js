import { noop } from 'lodash';
import React from 'react';
import { OverlayCompleted } from 'src/components';
import { uidEquals } from 'app/scripts/helpers';


describe('OverlayCompleted', () => {
  const createComponent = (props = {}) => (
    <OverlayCompleted
      title="For Those Of You Who Have Never (And Also Those Who Have)"
      {...props}
    />
  );

  it('should render', () => {
    const el = render(createComponent());
    expect(el).toMatchSnapshot();
  });

  it('should render with a dismiss button', () => {
    const el = shallow(createComponent({ onDismissPress: noop }));
    expect(el).toMatchSnapshot();
  });

  it('should call onDismissPress when dismiss button is pressed', () => {
    const onDismissPress = jest.fn();
    const el = shallow(createComponent({ onDismissPress }));

    el.findWhere(uidEquals('dismiss'))
      .simulate('press');

    expect(onDismissPress.mock.calls).toEqual([[]]);
  });
});
