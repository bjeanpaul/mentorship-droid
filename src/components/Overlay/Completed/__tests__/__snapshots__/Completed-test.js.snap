exports[`OverlayCompleted should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      Object {
        "backgroundColor": "rgba(246, 111, 63, 0.95)"
      }
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {}
      ]
    } />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 64
      }
    }>
    <Image
      source={1} />
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": undefined,
            "fontSize": 20
          }
        ]
      }>
      For Those Of You Who Have Never (And Also Those Who Have)
    </Text>
  </View>
</View>
`;

exports[`OverlayCompleted should render with a dismiss button 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function buildPredicate],
    "childrenOfNode": [Function childrenOfNode],
    "findWhereUnwrapped": [Function findWhereUnwrapped]
  },
  "length": 1,
  "node": Object {
    "$$typeof": Symbol(react.element),
    "_owner": null,
    "_store": Object {},
    "key": null,
    "props": Object {
      "children": Array [
        Object {
          "$$typeof": Symbol(react.element),
          "_owner": null,
          "_store": Object {},
          "key": null,
          "props": Object {
            "children": Object {
              "$$typeof": Symbol(react.element),
              "_owner": null,
              "_store": Object {},
              "key": null,
              "props": Object {
                "onPress": [Function noop],
                "type": 1,
                "uid": "dismiss"
              },
              "ref": null,
              "type": [Function HeaderIcon]
            },
            "style": undefined
          },
          "ref": null,
          "type": [Function Header]
        },
        Object {
          "$$typeof": Symbol(react.element),
          "_owner": null,
          "_store": Object {},
          "key": null,
          "props": Object {
            "children": Array [
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "source": 1
                },
                "ref": null,
                "type": [Function Image]
              },
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "children": "For Those Of You Who Have Never (And Also Those Who Have)",
                  "style": Object {
                    "color": undefined,
                    "fontSize": 20
                  }
                },
                "ref": null,
                "type": [Function Text]
              }
            ],
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 64
            }
          },
          "ref": null,
          "type": [Function View]
        }
      ],
      "style": Object {
        "backgroundColor": "rgba(246, 111, 63, 0.95)"
      }
    },
    "ref": null,
    "type": [Function BaseView]
  },
  "nodes": Array [
    Object {
      "$$typeof": Symbol(react.element),
      "_owner": null,
      "_store": Object {},
      "key": null,
      "props": Object {
        "children": Array [
          Object {
            "$$typeof": Symbol(react.element),
            "_owner": null,
            "_store": Object {},
            "key": null,
            "props": Object {
              "children": Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "onPress": [Function noop],
                  "type": 1,
                  "uid": "dismiss"
                },
                "ref": null,
                "type": [Function HeaderIcon]
              },
              "style": undefined
            },
            "ref": null,
            "type": [Function Header]
          },
          Object {
            "$$typeof": Symbol(react.element),
            "_owner": null,
            "_store": Object {},
            "key": null,
            "props": Object {
              "children": Array [
                Object {
                  "$$typeof": Symbol(react.element),
                  "_owner": null,
                  "_store": Object {},
                  "key": null,
                  "props": Object {
                    "source": 1
                  },
                  "ref": null,
                  "type": [Function Image]
                },
                Object {
                  "$$typeof": Symbol(react.element),
                  "_owner": null,
                  "_store": Object {},
                  "key": null,
                  "props": Object {
                    "children": "For Those Of You Who Have Never (And Also Those Who Have)",
                    "style": Object {
                      "color": undefined,
                      "fontSize": 20
                    }
                  },
                  "ref": null,
                  "type": [Function Text]
                }
              ],
              "style": Object {
                "alignItems": "center",
                "flex": 1,
                "paddingLeft": 24,
                "paddingRight": 24,
                "paddingTop": 64
              }
            },
            "ref": null,
            "type": [Function View]
          }
        ],
        "style": Object {
          "backgroundColor": "rgba(246, 111, 63, 0.95)"
        }
      },
      "ref": null,
      "type": [Function BaseView]
    }
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": Object {
        "$$typeof": Symbol(react.element),
        "_owner": null,
        "_store": Object {},
        "key": null,
        "props": Object {
          "onDismissPress": [Function noop],
          "title": "For Those Of You Who Have Never (And Also Those Who Have)"
        },
        "ref": null,
        "type": [Function CompletedOverlay]
      },
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onDismissPress": [Function noop],
          "title": "For Those Of You Who Have Never (And Also Those Who Have)"
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function enqueueCallback],
          "enqueueCallbackInternal": [Function enqueueCallbackInternal],
          "enqueueElementInternal": [Function enqueueElementInternal],
          "enqueueForceUpdate": [Function enqueueForceUpdate],
          "enqueueReplaceState": [Function enqueueReplaceState],
          "enqueueSetState": [Function enqueueSetState],
          "isMounted": [Function isMounted],
          "validateCallback": [Function validateCallback]
        }
      },
      "_mountOrder": 12,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": Object {
        "_currentElement": Object {
          "$$typeof": Symbol(react.element),
          "_owner": null,
          "_store": Object {},
          "key": null,
          "props": Object {
            "children": Array [
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "children": Object {
                    "$$typeof": Symbol(react.element),
                    "_owner": null,
                    "_store": Object {},
                    "key": null,
                    "props": Object {
                      "onPress": [Function noop],
                      "type": 1,
                      "uid": "dismiss"
                    },
                    "ref": null,
                    "type": [Function HeaderIcon]
                  },
                  "style": undefined
                },
                "ref": null,
                "type": [Function Header]
              },
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "children": Array [
                    Object {
                      "$$typeof": Symbol(react.element),
                      "_owner": null,
                      "_store": Object {},
                      "key": null,
                      "props": Object {
                        "source": 1
                      },
                      "ref": null,
                      "type": [Function Image]
                    },
                    Object {
                      "$$typeof": Symbol(react.element),
                      "_owner": null,
                      "_store": Object {},
                      "key": null,
                      "props": Object {
                        "children": "For Those Of You Who Have Never (And Also Those Who Have)",
                        "style": Object {
                          "color": undefined,
                          "fontSize": 20
                        }
                      },
                      "ref": null,
                      "type": [Function Text]
                    }
                  ],
                  "style": Object {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "paddingTop": 64
                  }
                },
                "ref": null,
                "type": [Function View]
              }
            ],
            "style": Object {
              "backgroundColor": "rgba(246, 111, 63, 0.95)"
            }
          },
          "ref": null,
          "type": [Function BaseView]
        },
        "_debugID": 2,
        "_renderedOutput": Object {
          "$$typeof": Symbol(react.element),
          "_owner": null,
          "_store": Object {},
          "key": null,
          "props": Object {
            "children": Array [
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "children": Object {
                    "$$typeof": Symbol(react.element),
                    "_owner": null,
                    "_store": Object {},
                    "key": null,
                    "props": Object {
                      "onPress": [Function noop],
                      "type": 1,
                      "uid": "dismiss"
                    },
                    "ref": null,
                    "type": [Function HeaderIcon]
                  },
                  "style": undefined
                },
                "ref": null,
                "type": [Function Header]
              },
              Object {
                "$$typeof": Symbol(react.element),
                "_owner": null,
                "_store": Object {},
                "key": null,
                "props": Object {
                  "children": Array [
                    Object {
                      "$$typeof": Symbol(react.element),
                      "_owner": null,
                      "_store": Object {},
                      "key": null,
                      "props": Object {
                        "source": 1
                      },
                      "ref": null,
                      "type": [Function Image]
                    },
                    Object {
                      "$$typeof": Symbol(react.element),
                      "_owner": null,
                      "_store": Object {},
                      "key": null,
                      "props": Object {
                        "children": "For Those Of You Who Have Never (And Also Those Who Have)",
                        "style": Object {
                          "color": undefined,
                          "fontSize": 20
                        }
                      },
                      "ref": null,
                      "type": [Function Text]
                    }
                  ],
                  "style": Object {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "paddingTop": 64
                  }
                },
                "ref": null,
                "type": [Function View]
              }
            ],
            "style": Object {
              "backgroundColor": "rgba(246, 111, 63, 0.95)"
            }
          },
          "ref": null,
          "type": [Function BaseView]
        }
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false
    },
    "getRenderOutput": [Function getRenderOutput],
    "render": [Function render]
  },
  "root": [Circular],
  "unrendered": Object {
    "$$typeof": Symbol(react.element),
    "_owner": null,
    "_store": Object {},
    "key": null,
    "props": Object {
      "onDismissPress": [Function noop],
      "title": "For Those Of You Who Have Never (And Also Those Who Have)"
    },
    "ref": null,
    "type": [Function CompletedOverlay]
  }
}
`;
